drivers/input/ajoystick.c

```diff
@@ -100,6 +100,7 @@ struct ajoy_open_s
    * driver events.
    */
 
+  bool ao_pollpending;
```

Comment:
Why this reside here and not in `struct btn_upperhalf_s`? I'm just thinking from the analogy with network sockets. If data arrive and there are many waiters, then only one can read the data and not all. Or I'm missing something?

Reply:
Input device is different from net device,  the broadcast semantics is normally used by the input device, which mean that multiple reader will receive the same event from one hardware device.